driver/src/main/scala/com/stratio/sparkta/driver/service/StreamingContextService.scala

```diff
@@ -165,30 +165,45 @@ object SparktaJob {
     })
   }
 
-  def rollups(apConfig: AggregationPoliciesDto,
-              operators: Seq[Operator],
-              dimensionsMap: Map[String, Dimension]): Seq[Rollup] =
-    apConfig.rollups.map(r => {
-      val components = r.dimensionAndPrecision.map(dab => {
-        dimensionsMap.get(dab.dimensionName) match {
+  private def getOutputsWithNames(outputs: Seq[Output], selectedOutputs: Seq[String]): Seq[Output] = {
+    outputs.filter(out => selectedOutputs.contains(out.getName))
+  }
+
+  def cubes(apConfig: AggregationPoliciesDto,
+            operators: Seq[Operator],
+            outputs: Seq[Output],
+            dimensionsMap: Map[String, Dimension]): Seq[Cube] =
+    apConfig.cubes.map(r => {
+      val name = r.name
+      val multiplexer = Try(r.multiplexer.toBoolean).getOrElse(false)
+      val components = r.dimensions.map(dab => {
+        dimensionsMap.get(dab.dimension) match {
           case Some(x: Dimension) => getDimensionPrecision(x, dab)
-          case None => throw new DriverException("Dimension name " + dab.dimensionName + " not found.")
+          case None => throw new DriverException("Dimension name " + dab.dimension + " not found.")
         }
       })
 
-      val operatorsForRollup = Option(r.operators) match {
+      val operatorsForCube = Option(r.operators) match {
```

Comment:
Please don't merge multiple refactors, functionalities,... in same PR. This things makes PR's get bigger